fe/src/app/advert/page.tsx

```diff
@@ -0,0 +1,184 @@
+import { Button, Input, Tab, TabPanel, Tabs, TabsBody, TabsHeader, Textarea } from "@/lib/materialTailwindExports";
+import { QuestionsFormType, SimpleFormType } from "@/types/CreateAdvertForm";
+import { FORM_BOX_SHADOW } from "@/utils/constants";
+import { QUESTIONS_FORM_TYPE, SIMPLE_FORM_TYPE, questionsFormDefaultValues, questionsFormSchema, simpleFormDefaultValues, simpleFormSchema } from "./utils";
+import { USER_ADVERTS_ROUTE } from "@/utils/urls";
+import ImageDropzone from "@/components/ImageDropzone/ImageDropzone";
+
+const AdvertPage = () => {
+  const [formType, setFormType] = useState<string>(SIMPLE_FORM_TYPE);
+  const [questionsFormData, setQuestionsFormData] = useState<QuestionsFormType>(questionsFormDefaultValues);
+  const [simpleFormData, setSimpleFormData] = useState<SimpleFormType>(simpleFormDefaultValues);
+
+  const handleEndDateChange = (date: DateValueType) => {
+    if (formType === SIMPLE_FORM_TYPE && date) {
+      setSimpleFormData((formData) => ({
+        ...formData,
+        endDate: date.endDate,
+      }));
+    } else if (date) {
+      setQuestionsFormData((formData) => ({
+        ...formData,
+        endDate: date.endDate,
+      }));
+    }
+  }
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const { name, value } = e.target;
+    if (formType === SIMPLE_FORM_TYPE) {
+      setSimpleFormData((formData) => ({
+        ...formData,
+        [name]: value,
+      }));
+    } else {
+      setQuestionsFormData((formData) => ({
+        ...formData,
+        [name]: value,
+      }));
+    }
+  };
+
+  const handleYesOrNoValue = (e: React.ChangeEvent<HTMLInputElement>, value: boolean) => {
+    const { name } = e.target;
+    setQuestionsFormData((formData) => ({
+      ...formData,
+      [name]: value,
+    }));
+  }
+
+  const handleImagesUpload = (imagesBase64Data: string[]) => {
+    if (formType === SIMPLE_FORM_TYPE) {
+      setSimpleFormData((formData) => ({
+        ...formData,
+        photos: imagesBase64Data
+      }));
+    } else {
+      setQuestionsFormData((formData) => ({
+        ...formData,
+        photos: imagesBase64Data
+      }));
+    }
+  }
+
+  const validateData = async (): Promise<boolean> => {
+    try {
+      if (formType === SIMPLE_FORM_TYPE) await simpleFormSchema.validate(simpleFormData, { abortEarly: false });
+      else await questionsFormSchema.validate(questionsFormData, { abortEarly: false });
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const isValid = await validateData();
+
+    if (isValid) {
+      const data = formType === SIMPLE_FORM_TYPE ? simpleFormData : questionsFormData;
+      const body = {
+        ...data,
+        isQuestionsFormType: formType,
+      }
+      const response = await fetch(`${process.env.NEXT_PUBLIC_API_SERVER_URL}${USER_ADVERTS_ROUTE}`, {
```

Comment:
This request still does not work, needs improvements for middleware or something to pass correct token and other headers.

Reply:
yup, It should work now if you pull my branch into yours

Reply:
let me know if it did start to work, just remove `mode: 'no-cors'` line as that's important

Reply:
No need for `mode: no-cors`, fixed request.